src/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -38,6 +38,34 @@ public int Start()
                 .ExitCode;
         }
 
+        public RunCommand(string configuration,
+            string framework,
+            bool noBuild,
+            string project,
+            IReadOnlyCollection<string> args)
+        {
+            Configuration = configuration;
+            Framework = framework;
+            NoBuild = noBuild;
+            Project = project;
+            Args = args;
+        }
+
+        public RunCommand MakeNewWithReplaced(string configuration = null,
```

Comment:
Args has a public setter, and the only thing you are changing with this MakeNewWithReplace is args. I don't think we need this right now.